src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -344,6 +344,30 @@ public function setAnnotatedClassCache(array $annotatedClasses)
     }
 
     /**
+     * Return a list of classes to be cached away when the container is build.
+     *
+     * If a class contains annotations, add it to getAnnotatedClassesToCompile()instead.
```

Comment:
Not sure this is worth a duplicated extra comment, the API looks explicit enough